packages/shared/src/components/modals/referral/GenericReferralModal.tsx

```diff
@@ -37,11 +38,19 @@ function GenericReferralModal({
       target_id: TargetId.GenericReferralPopup,
     }),
   });
+  const { trackEvent } = useAnalyticsContext();
   const onShareClick = () => {
     onShareOrCopyLink();
     setShareState(true);
   };
 
+  useEffect(() => {
+    trackEvent({
+      event_name: AnalyticsEvent.Impression,
+      target_type: TargetType.ReferralPopup,
+    });
+  }, [trackEvent]);
```

Comment:
For impression we always leave this empty for now. 
I know it has edge-cases but rather keep everything uniform with code base.

Reply:
Ah, right. To ensure we trigger it only once. Let me refactor.